### PR TITLE
 added bad request exception and json error check for request

### PR DESCRIPTION
**Pull request by RahulRavindren**
Please have a look at the code . If it's wrong or corrections to be done. Let me know.

### CODE REVIEW

File: src/lib/api.js [hunk 1]
```diff
@@ -28,6 +28,14 @@ var ApiBase = function (config) {
                 console.log('onComplete ajax call');
                 console.log(response.status);
                 console.log(response.statusText);
+				try{
```
Comment (mitechie) [inline src/lib/api.js:31]: this looks indented a bit too much. The indentation should be spaces and pass jslint. Can you double check it?

File: src/lib/api.js [hunk 2]
```diff
@@ -28,6 +28,14 @@ var ApiBase = function (config) {
...
+					if(response.status != 200) throw "bad";
```
Comment (mitechie) [inline src/lib/api.js:32]: please always use {} around if statements. It's a good safety measure.

File: src/lib/api.js [hunk 3]
```diff
@@ -28,6 +28,14 @@ var ApiBase = function (config) {
...
+					JSON.parse(response.json);
+				}catch(e){
```
Comment (mitechie) [inline src/lib/api.js:34]: this should have spaces around it

```
} catch (exc) {
```

File: src/lib/api.js [hunk 4]
```diff
@@ -28,6 +28,14 @@ var ApiBase = function (config) {
...
+					if(e == 'bad'|| e== 'SyntaxError')
```
Comment (mitechie) [inline src/lib/api.js:35]: another case for {} please

**Comment by mitechie** [comment]
Thanks so much! This looks like a solid change. I've noted a couple of small style nitpicks to look at please.

File: src/lib/api.js (hunk 1 above)
Comment (RahulRavindren) [inline src/lib/api.js:31]: thanks for letting me know will crosscheck and also pass jslint to it

**Comment by mitechie** [comment]
I adjusted this and have landed a version of this to make sure we can decode the JSON response else we call the error callback.

**Comment by RahulRavindren** [comment]
I am so sorry Rick. Couldn't look at this because of time
constraints.Anyways the tweak was needed indeed. I will make sure that my
involvement is there on this.

**Pull request merged**